tests/frame/to_pandas_test.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from typing import Any
+
+import pandas as pd
+import pytest
+
+import narwhals.stable.v1 as nw
+from narwhals.utils import parse_version
+
+
+@pytest.mark.filterwarnings("ignore:.*Passing a BlockManager.*:DeprecationWarning")
+@pytest.mark.skipif(
+    parse_version(pd.__version__) < parse_version("2.0.0"),
+    reason="too old for pandas-pyarrow",
+)
+def test_convert_pandas(constructor_with_pyarrow: Any) -> None:
+    data = {"a": [1, 3, 2], "b": [4, 4, 6], "z": [7.0, 8, 9]}
+    df_raw = constructor_with_pyarrow(data)
+    result = nw.from_native(df_raw).to_pandas()  # type: ignore[union-attr]
+
+    if constructor_with_pyarrow.__name__.startswith("pandas"):
+        expected = constructor_with_pyarrow(data)
+    else:
+        expected = pd.DataFrame(data)
```

Comment:
I am not too happy with this, but `asserting_frame_equal` clearly takes the backend into consideration. 
Alternative would be to just check for values

Reply:
looks good, not sure what else we can do here